### PR TITLE
Feature/panda Merge 3176 customer coupon component into PANDA-3170

### PR DESCRIPTION
**Pull request by Qingyu-Yang**

### CODE REVIEW

File: projects/core/src/occ/adapters/customer-coupon/occ-customer-coupon.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    return this.http
+      .delete(url, { headers })
+      .pipe(catchError((error: any) => throwError(error)));
+  }
+
+  turnOnNotification(
+    userId: string,
+    couponCode: string
+  ): Observable<CustomerCouponNotification> {
+    const url =
```
Comment (Jerify02) [inline projects/core/src/occ/adapters/customer-coupon/occ-customer-coupon.adapter.ts]: This url can be extracted a common url.

Comment (skyyuweiwen) [reply projects/core/src/occ/adapters/customer-coupon/occ-customer-coupon.adapter.ts]: Updated.

File: projects/core/src/user/store/reducers/customer-coupon.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  action: fromActions.CustomerCouponAction
+): CustomerCouponSearchResult {
+  switch (action.type) {
+    case fromActions.LOAD_CUSTOMER_COUPONS_SUCCESS: {
+      return action.payload ? action.payload : initialState;
+    }
+
+    case fromActions.SUBSCRIBE_CUSTOMER_COUPON_SUCCESS: {
+      const updatedCustomerCoupon = action.payload.coupon;
+      state.coupons.map((customerCoupon: CustomerCoupon) =>
```
Comment (Jerify02) [inline projects/core/src/user/store/reducers/customer-coupon.reducer.ts]: It seems to violate the pure function principle, should not modify the state directly.

File: projects/core/src/user/store/reducers/customer-coupon.reducer.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        customerCoupon.couponId === updatedCustomerCoupon.couponId
+          ? (customerCoupon.notificationOn = true)
+          : customerCoupon
+      );
+      return { ...state };
+    }
+
+    case fromActions.UNSUBSCRIBE_CUSTOMER_COUPON_SUCCESS: {
+      const updatedCouponCode = action.payload;
+      state.coupons.map((customerCoupon: CustomerCoupon) =>
```
Comment (Jerify02) [inline projects/core/src/user/store/reducers/customer-coupon.reducer.ts]: ditto.

File: projects/storefrontlib/src/cms-components/myaccount/my-coupons/coupon-card/coupon-card.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { CouponCardComponent } from './coupon-card.component';
+import { I18nTestingModule, CustomerCoupon } from '@spartacus/core';
+import { By } from '@angular/platform-browser';
+import { ModalService } from '@spartacus/storefront';
```
Comment (Jerify02) [inline projects/storefrontlib/src/cms-components/myaccount/my-coupons/coupon-card/coupon-card.component.spec.ts]: It probably should use relative path, such as `../../....`.

File: projects/storefrontlib/src/cms-components/myaccount/my-coupons/my-coupons.component.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    byStartDateDesc: 'startDate:desc',
+    byEndDateAsc: 'endDate:asc',
+    byEndDateDesc: 'endDate:desc',
+  };
+  sort = 'byStartDateAsc';
+  sortLabels = {
+    byStartDateAsc: 'Start Date (ascending)',
+    byStartDateDesc: 'Start Date (descending)',
+    byEndDateAsc: 'End Date (ascending)',
+    byEndDateDesc: 'End Date (descending)',
```
Comment (Jerify02) [inline projects/storefrontlib/src/cms-components/myaccount/my-coupons/my-coupons.component.ts]: These labels probably should be handled with localization?

File: projects/core/src/user/services/find-product-search-page-meta.resolver.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import { PageMeta, Page } from '../../cms/model/page.model';
+import { TranslationService } from '../../i18n/translation.service';
+import { PageType } from '../../model/cms.model';
+import { RoutingService } from '../../routing';
+import { ProductSearchService } from '../../product/facade/product-search.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FindProductSearchPageMetaResolver extends PageMetaResolver
```
Comment (Seven7hang) [inline projects/core/src/user/services/find-product-search-page-meta.resolver.ts]: FindProductsPageMetaResolver

**Pull request merged**